### DISCUSSION TITLE
Color effects support for android app

### DISCUSSION THREAD

**Discussion by TheEmpressMiaw** [awawa-dev/HyperHDR#154] [Q&A]
@awawa-dev 
Is there anyway I can get the color effects to show up in the [android app](https://play.google.com/store/apps/details?id=nl.hyperion.hyperionfree) ?
Right now the effects tab is empty (screenshot below)
![HyperHDR](https://user-images.githubusercontent.com/29691009/141694025-fe81b557-908a-4efa-bbb4-83b606e3daf0.jpg)

But when I connect to Hyperion.NG, i am able to toggle different color effects (screenshot below)
![Hyperion](https://user-images.githubusercontent.com/29691009/141694047-fbd43e6e-d2ad-4af5-82bd-31e29be6cdcb.jpg)

Because of this issue, I have both HyperHDR and Hyperion.NG installed on my system.
I have to stop HyperHDR and start Hyperion.NG when I want to toggle the effects.
And vice versa when I want to watch TV.

If I could gets the effects to show up in hyperHDR, then I could ditch the Hyperion.NG forever!

Any help will be appreciated, Thank you!

**Comment by awawa-dev** [comment]
Hi
That mobile app is abandon even by hyperion ng... long time ago without any update and since then a lot has changed (for example instances). Unfortunately have no clue how to make it work with hyperhdr...the easiest way is probably to create new one.
Regards
Awawa

**Comment by TheEmpressMiaw** [comment]
Hi, Thanks for replying!

There is a workaround by using [HTTP Request Shortcuts](https://play.google.com/store/apps/details?id=ch.rmy.android.http_shortcuts) android app and HyperHDR JSON API to toggle the effects.
The only problem is I have to manually add all combinations of color effects I use.
Is there any JSON API endpoint to list all the available color effects ?

**Comment by awawa-dev** [comment]
Yes, you can use ```serverinfo``` API command. In hyperhdr JSON API tab run 'Server resources & state information' to see how it works and what it returns. Effects array list is under ```info>effects``` node.

**Comment by TheEmpressMiaw** [comment]
The `serverinfo` API endpoint works well and it does list all the available color effects.
I guess, as you said, creating a new android app to make the REST API calls will be better and quicker than to fix the existing abandoned app
